Detectors/TPC/workflow/src/ZSSpec.cxx

```diff
@@ -106,31 +106,23 @@ DataProcessorSpec getZSEncoderSpec(std::vector<int> const& inputIds, bool zs10bi
           LOG(ERROR) << "sector header missing on header stack";
           return;
         }
-        const int& sector = sectorHeader->sector;
-        // check the current operation, this is used to either signal eod or noop
-        // FIXME: the noop is not needed any more once the lane configuration with one
-        // channel per sector is used
+        const int& sector = sectorHeader->sector();
         if (sector < 0) {
-          if (operation < 0 && operation != sector) {
-            // we expect the same operation on all inputs
-            LOG(ERROR) << "inconsistent lane operation, got "
-                       << sector << ", expecting " << operation;
-          } else if (operation == 0) {
-            // store the operation
-            operation = sector;
-          }
+          // this is a legacy condition, the EOS event is now handled by DPL
+          throw std::runtime_error("control events are not expected to be transmitted in the TPCSector header");
```

Comment:
If this is legacy, can't we remove it now?

Reply:
true, I did in the other case already